File: kotlin-core/src/test/java/maw/org/wedding/NavigationControllerTest.java
package maw.org.wedding;

import org.junit.Test;
import org.maw.wedding.navigation.Bus;
import org.maw.wedding.navigation.NavigationEvent;
import org.maw.wedding.navigation.NavigationEventController;
import org.maw.wedding.navigation.Navigator;
import org.maw.wedding.navigation.Screen;
import org.maw.wedding.navigation.ScreenEvent;
import org.mockito.Mockito;

import java.util.ArrayList;
import java.util.List;

import static org.mockito.Mockito.mock;

public class NavigationControllerTest {

    @Test
    public void testEventDispatched() {

        Navigator mockNavigator = Mockito.mock(Navigator.class);
        NavigationEventController navigationController = new NavigationEventController(mockNavigator, Mockito.mock(Bus.class));

        ScreenEvent screenEvent = new ScreenEvent(mock(Screen.class), 1);

        List<ScreenEvent> screenEvents = new ArrayList<>();
        screenEvents.add(screenEvent);

        navigationController.registerForScreenEvents(screenEvents);
        navigationController.onReady();

        NavigationEvent navigationEvent = new NavigationEvent(1);

        navigationController.onEvent(navigationEvent);

        Mockito.verify(mockNavigator).navigate(screenEvent);
    }



}

File: kotlin-core/src/test/kotlin/maw/org/wedding/NavigationControllerTest.kt
package maw.org.wedding

import maw.org.wedding.util.TestableNavigator
import maw.org.wedding.util.TestableScreen
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.maw.wedding.navigation.Event
import org.maw.wedding.navigation.NavigationEventBus
import org.maw.wedding.navigation.NavigationEventController
import org.maw.wedding.navigation.ScreenEvent
import java.util.*

class NavigationControllerTest {

    lateinit var navigator: TestableNavigator

    @Before
    fun setup() {
        navigator = TestableNavigator()
    }

    @Test
    fun testEventDispatched() {

        val navigationController = NavigationEventController(navigator, NavigationEventBus())

        val screenEvent = ScreenEvent(TestableScreen(), 1)

        val screenEvents = ArrayList<ScreenEvent>()
        screenEvents.add(screenEvent)

        navigationController.registerForScreenEvents(screenEvents)
        navigationController.onReady()

        val navigationEvent = Event(1)

        navigationController.onEvent(navigationEvent)

        Assert.assertEquals(navigator.screenEvent, screenEvent)
    }


}
